src/mixxx.cpp

```diff
@@ -217,6 +217,15 @@ MixxxMainWindow::MixxxMainWindow(QApplication* pApp, const CmdlineArgs& args)
     mixxx::Translations::initializeTranslations(
         m_pSettingsManager->settings(), pApp, args.getLocale());
 
+#ifdef __LINUX__
+    // If the desktop features a global menubar and we go fullscreen during
+    // startup, move the menubar to the window like in slotViewFullScreen()
+    bool fullscreenPref = m_pSettingsManager->settings()->getValue<bool>(
+            ConfigKey("[Config]", "StartInFullscreen"));
+    QApplication::setAttribute(
+            Qt::AA_DontUseNativeMenuBar,
+            args.getStartInFullscreen() || fullscreenPref);
+#endif // __LINUX__
```

Comment:
Note for future improvements:
we can test for this attribute _before_ setting it, then all recreating and reconnecting can be limited to platforms that actually require/support it.